### PR TITLE
⚠️ Deprecates the auto configure functionality

### LINKED ISSUE #1899: Remove the tagging functionality of the VSphereFailureDomain API
/kind feature

**Describe the solution you'd like**
Traditionally VI admins have been responsible for tagging and managing the vCenter resources. Having CAPV tag resources out of band is a departure from this behavior. From the feedback we received from our users, this is a functionality that they do not want to expose to CAPV, but have the VI admins perform resource tagging as a pre-requisite. 

**Anything else you would like to add:**
Deprecate the functionality via the following steps:
- Introduce the deprecation notice on the API type
- Remove the controller logic for resource tagging

**Environment:**

- Cluster-api-provider-vsphere version: 
- Kubernetes version: (use `kubectl version`): 
- OS (e.g. from `/etc/os-release`):

### PR DESCRIPTION
**Pull request by srm09**
**What this PR does / why we need it**:
Deprecates the auto configure functionality from the VSphereDeploymentZone API.

**Which issue(s) this PR fixes**:
Fixes #1899 

**Special notes for your reviewer**:
This would also lead to removing some permissions from the vCenter user in use by CAPV.

**Release note**:
```release-note
VSphereDeploymentZone: Deprecates the auto configure functionality
```

### CODE REVIEW

**Comment by srm09** [comment]
/retitle ⚠️ Deprecates the auto configure functionality

File: apis/v1beta1/vspherefailuredomain_types.go [hunk 1]
```diff
@@ -53,6 +53,8 @@ type FailureDomain struct {
 	// TagCategory is the category used for the tag
 	TagCategory string `json:"tagCategory"`
 
+	// Deprecated: This will be released in a future release. Setting this value
```
Comment (yastij) [inline apis/v1beta1/vspherefailuredomain_types.go:56]: removed*

File: apis/v1beta1/vspherefailuredomain_webhook.go [hunk 2]
```diff
@@ -80,14 +76,3 @@ func (r *VSphereFailureDomain) ValidateUpdate(old runtime.Object) error {
 func (r *VSphereFailureDomain) ValidateDelete() error {
 	return nil
 }
-
-// Default implements webhook.Defaulter so a webhook will be registered for the type.
-func (r *VSphereFailureDomain) Default() {
```
Comment (yastij) [inline apis/v1beta1/vspherefailuredomain_webhook.go:85]: why are we removing the defaulting. This is just the deprecation announcement

Comment (srm09) [reply apis/v1beta1/vspherefailuredomain_webhook.go:85]: The broader idea behind the PR was:
1. To add the depreciation notice to the API and
2. To remove the functionality exercised when setting the flag 

According to the PR, irrespective of what value the user sets, no resource tagging gets executed.

**Comment by sbueringer** [comment]
@srm09 @yastij 

Discussed this in the offce hours yesterday.

We think it might be good to do a cycle with only deprecation first to see if we get user feedback that this feature is still used and then remove it in a subsequent release. So we would suggest to only deprecate the field in this PR.

Does this sound reasonable to you?

**Comment by srm09** [comment]
@sbueringer Not objected to the deprecation of the field before we yank out the functionality entirely. Feel free to open a separate PR, and close this one.

**Comment by srm09** [comment]
/close
@sbueringer to open another one with just the deprecation piece.

**Pull request closed without merging**